### ISSUE TITLE
Agents stuck thinking 

### ISSUE THREAD

**Issue by GideonLangenhoven** [TransformerOptimus/SuperAGI#221]
agents dont move passed the thinking phase even after 3 hours of deploy

**Comment by stevejford** [comment]
Same issue here 2

**Comment by SpeedoPasanen** [comment]
Same, all I get is "Thinking" with either GPT-4 and 3.5.

**Comment by neelayan7** [comment]
This is a common issue we're facing. We're releasing a patch for this along with the v0.0.4 release scheduled for today. Hopefully, it'll work then.

**Comment by stevejford** [comment]
> This is a common issue we're facing. We're releasing a patch for this along with the v0.0.4 release scheduled for today.

I looked at the logs and says something about some error next.js. Not sure if its related

**Comment by neelayan7** [comment]
> > This is a common issue we're facing. We're releasing a patch for this along with the v0.0.4 release scheduled for today.
> 
> I looked at the logs and says something about some error next.js. Not sure if its related

Can you share the logs?

**Comment by manzonif** [comment]
Any details about the patch? I have v0.0.4 installed, but I am now at third hour of thinking phase.
How should these situations be handled? In the GUI I only see the pause button.

`superagi-super__redis-1     | 1:M 08 Jun 2023 07:17:54.018 * 1 changes in 3600 seconds. Saving...
superagi-super__redis-1     | 1:M 08 Jun 2023 07:17:54.019 * Background saving started by pid 21
superagi-super__redis-1     | 21:C 08 Jun 2023 07:17:54.035 * DB saved on disk
superagi-super__redis-1     | 21:C 08 Jun 2023 07:17:54.035 * superagi-super__redis-1     | 1:M 08 Jun 2023 07:17:54.018 * 1 changes in 3600 seconds. Saving...
superagi-super__redis-1     | 1:M 08 Jun 2023 07:17:54.019 * Background saving started by pid 21
superagi-super__redis-1     | 21:C 08 Jun 2023 07:17:54.035 * DB saved on disk
superagi-super__redis-1     | 21:C 08 Jun 2023 07:17:54.035 * Fork CoW for RDB: current 0 MB, peak 0 MB, average 0 MB
superagi-super__redis-1     | 1:M 08 Jun 2023 07:17:54.119 * Background saving terminated with success
superagi-super__redis-1     | 1:M 08 Jun 2023 07:17:54.119 * Background saving terminated with success`

**Comment by trapperl** [comment]
I had a problem with it being stuck thinking and noticed an error.

`error_code=invalid_api_key error_message= error_param=None error_type=invalid_request_error message='OpenAI API error received' stream_error=False`

There was another issue that pointed me here

superagi\llms\openai.py

I found line 33 commented out

`# openai.api_key = get_config("OPENAI_API_KEY")`

I uncommented it and tried running it again.

Seems to work.

I have no idea what I'm doing and this might be a bad idea, but it worked.

**Comment by AxelAlinsky** [comment]
> I also got the same encountered the same issue be un-commenting the line in the openai.py file didn't work. However, What worked for me though is when utilizing the payment on Openai. I guess you have to start paying.....

**Comment by luciferlinx101** [comment]
@trapperl Hey in the current version we added an option to add key from frontend did you try that?

**Comment by neelayan7** [comment]
@GideonLangenhoven are you still facing the same issue with the newer version releases?

**Comment by iAbhinav** [comment]
Facing this with the latest release as well. Just cloned the repo.

**Comment by neelayan7** [comment]
Can you share some detailed logs from docker?

**Comment by kierglover** [comment]
I'm also experiencing the same issue mentioned by @iAbhinav above. Here's the current logs from my terminal, not sure if that helps you out @neelayan7 

`superagi-proxy-1            | 172.18.0.1 - - [28/Jun/2023:13:52:23 +0000] "GET /api/agentexecutionfeeds/get/execution/1 HTTP/1.1" 200 4874 "http://localhost:3000/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36" "-"
superagi-backend-1          | INFO:     172.18.0.7:49550 - "GET /agentexecutionfeeds/get/execution/1 HTTP/1.0" 200 OK`

**Comment by neelayan7** [comment]
@iAbhinav can you please share the agent goals and description along with Model you're trying with, so that we can replicate this issue at our end and provide a solution?

**Comment by neelayan7** [comment]
@kierglover can you check if there are any error messages in the log?

**Comment by neelayan7** [comment]
Also, please check if you're OpenAI API keys are setup properly. That can be one of the reasons. Also, if you're using GPT-4 model, kindly check if you have GPT-4 access.

**Comment by kierglover** [comment]
Hi @neelayan7 thanks for the reply, indeed it seemed to be regarding trying to use GPT-4. When I switched to GPT3.5 it worked well.

**Comment by neelayan7** [comment]
Awesome. Closing this issue as this is solved. Please reopen if you face any issue related to this.

**Comment by gemoo23** [comment]
Hello. I have installed superAGI to 2 different computers and theya re stuck at thinking. I have a paid membership to chatGPT. I double checked my openAI API key and it is correct. When I check the usage on openAI API it shows 0 so the agent is not calling the API.
Is there a way to debug the code with visual studio code to see where it is getting stuck? Thanks.

**Comment by gemoo23** [comment]
I had to create a paid account for openAI besides the chatGPT subscription. Created a new API key and now it works. Thanks.

**Comment by neelayan7** [comment]
Oh awesome!

**Comment by nickgrantmedia** [comment]
Hey everyone! I'm pretty new at this. I installed everything by following this video: https://youtu.be/bCgFgUSCEII?si=okjjlvkWB_DzToIY
I then got the thinking error where it couldn't find the tool. After googling a bunch I realized it was my openai API key and that I needed to enter billing info. I did that and then it started to work and completed some of the goals, but now it is stuck on 'thinking' again for the last 30 minutes. Any other ideas on how to troubleshoot this? I don't have much technical or CS knowledge (I have a marketing background), so any input would be appreciated. Thanks!
